prov/shm/src/smr_init.c

```diff
@@ -100,6 +118,8 @@ static int smr_getinfo(uint32_t version, const char *node, const char *service,
 			cur->ep_attr->max_order_waw_size = 0;
 			cur->ep_attr->max_order_war_size = 0;
 		}
+		if (ptrace)
+			cur->ep_attr->max_msg_size = SMR_INJECT_SIZE;
```

Comment:
The use of ptrace as a bool here looks opposite from what the code is doing.  ptrace is true, so don't use it?

Reply:
Yeah, CMA requires ptrace so this is ambiguous. Maybe call it ptrace_scope and  fallback if it is at a non-zero level, or just move it to a boolean function like Sean suggests and fallback if the scope is restrictive.